ls1intum/Artemis
# `Development`: Reduce payload when sending result updates

### Checklist
#### General
#### Server

### Motivation and Context
To save bandwidth and improve performance we should not send unnecessary information using the websocket connections.

### Description
We recently changed the websocket server to send messages in a separate thread. For PE result notifications, we set values to null that are not needed for the student, and reconnect the values again after sending the message. After the introduction of multithreading, these values were directly connected again before sending the actual message, leading to a bigger payload than expected.

I assume that this also fixes #6981 (feedback gets added before sending in another thread, making it visible to the student)
### Steps for Testing
Push a new result for a programming exercise. Verify that Artemis automatically updates the build status and shows the new result.

#### Exam Mode Testing
Check that new results are automatically visible while working on a exam.

### Review Progress

#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Screenshots
before: 
![grafik](https://github.com/ls1intum/Artemis/assets/26540346/a79028c1-3abe-4863-b68c-cafc3d7262c5)

now: 
![grafik](https://github.com/ls1intum/Artemis/assets/26540346/88d886e3-f8b2-43d3-8509-a5580ec379c9)
